src/utils/searchQuery.js

```diff
@@ -0,0 +1,29 @@
+import { regionLabelToValue } from './region';
+import { cloudProviderLabelToValue } from './cloudProvider';
+import { statusLabelToValue } from './status';
+
+export function filtersToSearchQuery(filters) {
+  const searchCategories = Object.keys(filters);
+  const queries =
+    searchCategories
+      .map((searchCategory) => {
+        const searchValues = filters[searchCategory];
+        const searchCategoryResult = searchValues
+          .map((searchValue) => {
+            // Use the value the API needs rather than the human readable UI value
+            let modifiedSearchValue = searchValue;
+            if (searchCategory === 'cloud_provider') {
+              modifiedSearchValue = cloudProviderLabelToValue(searchValue);
+            } else if (searchCategory === 'region') {
+              modifiedSearchValue = regionLabelToValue(searchValue);
+            } else if (searchCategory === 'status') {
+              modifiedSearchValue = statusLabelToValue(searchValue);
+            }
+            return `${searchCategory} = ${modifiedSearchValue}`;
+          })
+          .join(' or ');
+        return `(${searchCategoryResult})`;
```

Comment:
Not sure if it is possible or not, but if you have a search key/value where the value is an empty array, this inner `map` will return `"()"`, which could lead to a search query like `"(name = sc-test-1) and ()"`.

Do you know if the backend API will handle that correctly? It might be worth filtering out potential empty values here just in case a future code change (FE or BE) causes a query like this to result in a bug.

Reply:
Technically, the `InstancesToolbarSearchFilter` component will make sure to just remove the field if the value ends up being an empty array (ie. `{ region: [] }`. That was a bug that was fixed with these changes. But you raise a good point though. If you look at just this `filtersToSearchQuery` function, we could make it more robust by preventing that from happening in case anything outside the function happens to change in the future.

Reply:
I've added a filter to remove anything with empty values. So something like
```
{
    name: ['sc-test-1'],
    owner: [],
}
 ```
 would produce
 ```
 (name = sc-test-1)
 ```